src/webapp/components/survey/SurveyForm.tsx

```diff
@@ -71,30 +72,50 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
         }
     };
 
+    useEffect(() => {
+        if (!loading) {
```

Comment:
we know that any section update is triggered on question update, so why have the section update logic in useEffect? I suspect that is what is triggering the react-hooks/exhaustive-deps error. Any section updation logic should be taken care of in updateQuestion. 
We could pre-process the list of questions that are effected by a given question update and save it in Questionnaire